### DISCUSSION TITLE
Configure cspell filetype

### DISCUSSION THREAD

**Discussion by justinhj** [mason-org/mason.nvim#745] [Q&A] [answered]
Is it possible to configure the cspell linter at all? I would like to at least turn it on for specific file types only. Or more advanced turn it on for comments with code but not for functions and variables.

**Comment by getlost01** [comment, marked as answer]
Yes, it is possible to configure the cspell linter to lint specific file types only or to lint specific parts of a file (e.g., comments, functions, variables).

To configure cspell to lint specific file types only, you can use the --files option and specify a list of file patterns to lint. For example, to lint only JavaScript and TypeScript files, you can use the following command:

`cspell --files "*.js,*.ts"`

To configure cspell to lint specific parts of a file, you can use the --files option and specify a list of file patterns to lint, followed by the --language option and the language identifier for the parts of the file you want to lint. For example, to lint only comments in JavaScript files, you can use the following command:

`cspell --files "*.js" --language "javascript.comments"`
You can find a list of supported language identifiers and the parts of a file they correspond to in the cspell documentation.

**Reply by justinhj** [reply]
Wonderful, thanks!
